prime-router/src/test/kotlin/fhirengine/engine/fhirRouterTests/FHIRBundleHelpersTests.kt

```diff
@@ -688,4 +693,136 @@ class FHIRBundleHelpersTests {
+        val bundle = messages[0]
+        assertThat(bundle).isNotNull()
+
+        // create the hl7 reader
+        val hl7Reader = HL7Reader(actionLogger)
+        // get the hl7 from the blob store
+        val hl7Message = File("src/test/resources/fhirengine/engine/hl7_with_birth_time.hl7").readText()
+        val hl7messages = hl7Reader.getMessages(hl7Message)
+
+        bundle.handleBirthTime(hl7messages[0])
+
+        val patient = FhirPathUtils.evaluate(
+            CustomContext(bundle, bundle),
+            bundle,
+            bundle,
+            "Bundle.entry.resource.ofType(Patient)"
+        )[0] as Patient
+
+        assertThat(patient).isNotNull()
+        val birthDateTimeValue = patient.birthDateElement.extension[0].value.primitiveValue()
+        assertThat(
+            birthDateTimeValue
+        ).isEqualTo("2023-05-04T13:10:23-05:00")
+    }
+
+    @Test
+    fun `Test birth date time no patient`() {
+        // set up
+        val actionLogger = ActionLogger()
+        val fhirBundle = File("src/test/resources/fhirengine/engine/fhir_without_patient.fhir").readText()
+        val messages = FhirTranscoder.getBundles(fhirBundle, actionLogger)
+        assertThat(messages).isNotEmpty()
+        val bundle = messages[0]
+        assertThat(bundle).isNotNull()
+
+        // create the hl7 reader
+        val hl7Reader = HL7Reader(actionLogger)
+        // get the hl7 from the blob store
+        val hl7Message = File("src/test/resources/fhirengine/engine/hl7_with_birth_time.hl7").readText()
+        val hl7messages = hl7Reader.getMessages(hl7Message)
+
+        bundle.handleBirthTime(hl7messages[0])
+
+        val patient = FhirPathUtils.evaluate(
+            CustomContext(bundle, bundle),
+            bundle,
+            bundle,
+            "Bundle.entry.resource.ofType(Patient)"
+        )[0] as Patient
+
+        assertThat(patient).isNotNull()
+        val birthDateTimeValue = patient.birthDateElement.extension[0].value.primitiveValue()
+        assertThat(
+            birthDateTimeValue
+        ).isEqualTo("2023-05-04T13:10:23-05:00")
+    }
+
+    @Test
+    fun `Test enhance bundle metadata 2-5-1`() {
+        // set up
+        val actionLogger = ActionLogger()
+        val fhirBundle = File("src/test/resources/fhirengine/engine/fhir_without_patient.fhir").readText()
+        val messages = FhirTranscoder.getBundles(fhirBundle, actionLogger)
+        assertThat(messages).isNotEmpty()
+        val bundle = messages[0]
+        assertThat(bundle).isNotNull()
+
+        // create the hl7 reader
+        val hl7Reader = HL7Reader(actionLogger)
+        // get the hl7 from the blob store
+        val hl7Message = File("src/test/resources/fhirengine/engine/hl7_with_birth_time.hl7").readText()
+        val hl7messages = hl7Reader.getMessages(hl7Message)
+
+        bundle.enhanceBundleMetadata(hl7messages[0])
+
+        assertThat(bundle.timestamp.toString()).isEqualTo("Wed Feb 10 17:07:37 PST 2021")
```

Comment:
We should avoid using toString for these kinds of comparisons; even if the dates are equivalent, it could result in failed tests depending on the timezone of the system it is run on. Either testing for equality to a second `Date`, or using `toGMTString` instead should work.

Reply:
Changed this to compare to a `Date`